### PR TITLE
Clean up undecodable 2019 Kusama Locks

### LINKED ISSUE #104: Bad storage key value in Kusama runtime
There are number of undecoable key value pairs in Kusama. Here are some from Balances.locks

- Key `0xc2261276cc9d1f8598ea4b6a74b15c2f218f26c73add634897550b4003b26bc625dd314d5cf9255ed0f3bdf2589adc292f47aed28c0690f788ed2dbceb1fc3ff`
- Value `0x087374616b696e672000000000000000000000000000000000ffffffff1f64656d6f63726163ffffffffffffffffffffffffffffffffc0b0070002`

- Key `0xc2261276cc9d1f8598ea4b6a74b15c2f218f26c73add634897550b4003b26bc6386a9e3f0e74067eef90a9a281c234e05d6fa9c4a25b20fd03913b81d7fccf13`
- Value `0x047374616b696e672000000000000000000000000000000000ffffffff1f`

- Key `0xc2261276cc9d1f8598ea4b6a74b15c2f218f26c73add634897550b4003b26bc6402fc69a7eb122806b64c9e27735e45009aa922e1ba4675758c75607a96a2172`
- Value `0x047374616b696e672000000000000000000000000000000000ffffffff1f`

- Key `0xc2261276cc9d1f8598ea4b6a74b15c2f218f26c73add634897550b4003b26bc645c200adc666cca26df86e2e4a5aa1af186606b9e2a5b86c14cc954228a3cd00`
- Value `0x047374616b696e672000000000000000000000000000000000ffffffff1f`

- Key `0xc2261276cc9d1f8598ea4b6a74b15c2f218f26c73add634897550b4003b26bc651e634907ab2f0b63eadc4091e366f51bcf6f73763f216dcf716cfe7757813e7`
- Value `0x0464656d6f63726163ffffffffffffffffffffffffffffffffc0b0070002`

- Key `0xc2261276cc9d1f8598ea4b6a74b15c2f218f26c73add634897550b4003b26bc65abb1b03f39287c41885312b50a18dc337de40d15f99c182eba0e5aa6629b9bc`
- Value `0x0464656d6f63726163ffffffffffffffffffffffffffffffff7218030002`

all of the above keys are touched by this block https://kusama.subscan.io/block/20466150?tab=event

### PR DESCRIPTION
**Pull request by liamaharon**
Closes #104. See issue for info.

### Migration logs
Screenshot 2023-12-12 at 12 10 58

### CODE REVIEW

**Comment by xlc** [comment]
While we can use migration to delete the storages, but should we just make a public referendum and use system.killStorage?

**Comment by bkchr** [comment]
I would say, both are equally fine. As we are not in any hurry to delete them, I think we can just keep the migration.

**Comment by liamaharon** [comment]
There are more undecodable keys. Converting to draft while I add them.

**Comment by bkchr** [comment]
Given the amount of keys, using `kill_storage` should be really much easier.

**Comment by liamaharon** [comment]
> Given the amount of keys, using `kill_storage` should be really much easier.

Do you mean resolve this with a sudo call instead of runtime upgrade?

File: relay/kusama/src/lib.rs [hunk 1]
```diff
@@ -1676,10 +1676,110 @@ pub type Migrations = migrations::Unreleased;
...
+						},
+					};
+
+					log::info!("Killing Lock {:?}", lock);
+					writes.saturating_inc();
+					frame_support::storage::unhashed::kill(&combined_key);
+					log::info!("Killed");
+				}
+
+				Weight::from_parts(reads, writes)
```
Comment (liamaharon) [inline relay/kusama/src/lib.rs:1754]:
```suggestion
				<Runtime as frame_system::Config>::DbWeight::get().reads_writes(reads, writes)

```

**Comment by bkchr** [comment]
> Do you mean resolve this with a sudo call instead of runtime migration?

Yes

**Comment by liamaharon** [comment]
Closing in favour of fixing with referenda https://kusama.subsquare.io/referenda/334

**Pull request closed without merging**